flocker/node/agents/test/test_blockdevice.py

```diff
@@ -3465,3 +3465,97 @@ def test_get_device_path_until_detach(self):
 
         self.assertRaises(UnattachedVolume,
                           self.cache.get_device_path, attached_id1)
+
+
+class GetDeviceForDatasetIdTests(SynchronousTestCase):
+    """
+    Tests for ``get_device_for_dataset_id``.
+    """
+
+    def test_found(self):
+        """
+        Filesystems created with ``CreateFilesystem`` can be looked up by
+        dataset ID using ``get_device_for_dataset_id``.
+        """
+        one_dataset_id = uuid4()
+        another_dataset_id = uuid4()
+        uninitialized_dataset_id = uuid4()
+
+        deployer = create_blockdevicedeployer(self)
+        api = deployer.block_device_api
+
+        # Here are a couple volumes that will have filesystems that the
+        # implementation will have to correctly identify.
+        one_volume = api.create_volume(
+            one_dataset_id, REALISTIC_BLOCKDEVICE_SIZE
```

Comment:
These and other ``IStateChange`` y tests use a loopback block device API and they mostly use `LOOPBACK_MINIMUM_ALLOCATABLE_SIZE` instead of `REALISTIC_BLOCKDEVICE_SIZE`.

So could you.